aten/src/ATen/native/TensorShape.cpp

```diff
@@ -392,6 +390,90 @@ Tensor cat(TensorList tensors, int64_t dim) {
   return result;
 }
 
+Tensor block_diag(TensorList tensors) {
+  Tensor result;
+  if (tensors.size() == 0) {
+    result = at::empty({0});
+    return result;
+  }
+
+  // Use at::result_type to find the right tensor to copy options from, so
```

Comment:
nit: Is it possible to use `native::result_type` https://github.com/pytorch/pytorch/blob/246416ac3b97c29c10a1221a71cce7e84ed8fda0/aten/src/ATen/native/TypeProperties.h#L13 to compute the result type instead? It looks like it is doing the same thing but I'm not sure why it isn't exposed in the `at::` namespace